libs/wingc/src/type_check.rs

```diff
@@ -6288,6 +6253,76 @@ impl<'a> TypeChecker<'a> {
 			(None, None)
 		}
 	}
+
+	fn type_check_lift_statement(&mut self, lift_quals: &ExplicitLift, env: &mut SymbolEnv) {
+		// TODO
+		// 1. Validate all obj are references to preflight objects
+		// 2. Validate all ops are inflight memebers of the objs
+		// 3. Disabled qual erros for inner block (in lifting.rs)
+		// 4. Type check the inner block
+
+		for qual in lift_quals.qualifications.iter() {
+			let (obj_type, obj_phase) = self.type_check_exp(&qual.obj, env);
+			// Skip unknown references (diagnotics already emitted in `resolve_reference`)
+			if obj_type.is_unresolved() {
+				continue;
+			}
+			// Make sure the object isn't inflight
+			if obj_phase == Phase::Inflight {
+				self.spanned_error(
+					&qual.obj,
+					format!("Expected a preflight object, but found {obj_phase} expression instead"),
+				);
+			}
+			// Make sure the object type is a preflight type
+			if !obj_type.is_preflight_object_type() {
+				self.spanned_error(
+					&qual.obj,
+					format!("Expected a preflight object type, but found {obj_type} instead"),
```

Comment:
(nit) I feel like "preflight object type" could be confusing. Maybe we can mention something in the diagnostic as a hint, or reword this in some way